app/models/grouping.rb

```diff
@@ -749,17 +749,16 @@ def marking_state(result)
   end
 
   def get_total_test_script_marks
-    total = 0
 
     #find the unique test scripts for this submission
-    test_script_ids = test_script_results.pluck(:id).uniq
+    test_script_ids = test_script_results.pluck(:test_script_id).uniq
 
     #add the latest result from each of our test scripts
     test_script_ids.sum do |test_script_id|
-      last_mark = self.test_script_results
-                      .where(test_script_id: test_script_id)
-                      .last
-      last_mark.nil? ? 0 : last_mark.marks_earned
+      last_result = self.test_script_results
+                        .where(test_script_id: test_script_id)
+                        .first
```

Comment:
1. There is an explicit order set for the association and it's DESC, so we need to use first to get the latest. I don't know why it's that way though, I can change it but I didn't want to risk affecting other queries.
2. Right, I don't think I got rid of it on purpose, I'll put it back.